snowplow_tracker/snowplow.py

```diff
@@ -0,0 +1,108 @@
+# """
+#     snowplow.py
+
+#     Copyright (c) 2013-2022 Snowplow Analytics Ltd. All rights reserved.
+
+#     This program is licensed to you under the Apache License Version 2.0,
+#     and you may not use this file except in compliance with the Apache License
+#     Version 2.0. You may obtain a copy of the Apache License Version 2.0 at
+#     http://www.apache.org/licenses/LICENSE-2.0.
+
+#     Unless required by applicable law or agreed to in writing,
+#     software distributed under the Apache License Version 2.0 is distributed on
+#     an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+#     express or implied. See the Apache License Version 2.0 for the specific
+#     language governing permissions and limitations there under.
+
+#     Authors: Jack Keene, Anuj More, Alex Dean, Fred Blundun, Paul Boocock
+#     Copyright: Copyright (c) 2013-2022 Snowplow Analytics Ltd
+#     License: Apache License Version 2.0
+# """
+import logging
+from typing import  Optional
+from snowplow_tracker import Tracker, Emitter, subject
+from snowplow_tracker.typing import JsonEncoderFunction
+
+# Logging
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+    
+
+"""
+Snowplow Class
+"""
+class Snowplow:
+
+    def __init__(
+        self,
+        trackers= {}) -> None:
+        """
+            :param trackers:         Dictionary of initialized trackers 
+            :type  trackers:         Dictionary | None  
+        """
+    
+        self.trackers = trackers
+    
+    def create_tracker(self, 
```

Comment:
I think we could make it even easier to use this method by following a similar pattern like we do in the Java or mobile trackers. Those trackers have configuration classes (e.g., `NetworkConfiguration`, `EmitterConfiguration`, `SubjectConfiguration`) which the users use to describe the configuration that they want and all the implementation details such as creating a `Tracker` or `Emitter` instance are handled on the background.

I imagine it could work something like this from the user point of view:

```python
emitter_config = EmitterConfiguration()
emitter_config.buffer_size = 100

subject = Subject() # we don't need to provide a configuration class for Subject I think
subject.set_user_id("uid")

Snowplow.create_tracker(
   namespace="ns",
   endpoint="https://...",
   app_id="app1",
   emitter_config=emitter_config,
   subject=subject)
```

I think it would be enough to abstract the Emitter implementations by creating an `EmitterConfiguration` class that contain the configuration for creating an Emitter. No need for `TrackerConfiguration` and `NetworkConfiguration` as we can have their properties accepted directly in the `create_tracker` method.

You can get inspiration from [Java](https://github.com/snowplow/snowplow-java-tracker/blob/master/src/main/java/com/snowplowanalytics/snowplow/tracker/Snowplow.java), [C++](https://github.com/snowplow/snowplow-cpp-tracker/blob/master/include/snowplow/snowplow.cpp), or [Rust](https://github.com/snowplow/snowplow-rust-tracker/blob/main/src/snowplow.rs) which has a very simple implementation currently.

Not sure if this makes sense, let me know and we can chat about it!

Reply:
Btw, the `EmitterConfiguration` should be an optional parameter, it should be able to create the tracker with default configuration if it's not passed.